CRM/Contact/BAO/Relationship.php

```diff
@@ -1865,7 +1865,8 @@ public static function getPermissionedContacts($contactID, $relTypeId = NULL, $n
 FROM civicrm_relationship cr, civicrm_contact cc, civicrm_relationship_type crt
 WHERE
 cr.contact_id_a         = %1 AND
-cr.is_permission_a_b    = 1 AND
+cr.is_permission_a_b    = 1 OR
+cr.is_permission_b_a    = 1 AND
```

Comment:
seems like the above OR clause should be enclosed in a bracket?

Reply:
@jitendrapurohit Any specific reason behind doing so ?

Reply:
Currently, If I apply this change and create a normal relationship between A and B so that A has permission to view and update B, calling this function for A returns all the contact ids present in the database. i.e, -

    $permissionedContactIds = CRM_Contact_BAO_Relationship::getPermissionedContacts($contact_id_a);
      
`$permissionedContactIds` has all the contact ids instead of holding only B's contact id that A has permission to.

Before this patch - it correctly returns a single contact id.

Reply:
@jitendrapurohit will check this ASAP